src/Http/Headers/src/ContentDispositionHeaderValue.cs

```diff
@@ -764,7 +764,7 @@ internal static byte UnEscapeAscii(char digit, char next)
                 || ((digit >= 'A') && (digit <= 'F'))
                 || ((digit >= 'a') && (digit <= 'f'))))
             {
-                throw new ArgumentException();
+                throw new ArgumentOutOfRangeException(nameof(digit));
```

Comment:
I'm guessing this isn't something that a developer could trigger through any normal coding pattern. If it is, then maybe we should have a more descriptive exception message, since `digit` is an internal parameter name that the developer didn't supply themselves and can't know what it means without reading our sources.

Assuming this isn't a mainstream scenario, then fine, I'm sure this suffices already!

Reply:
From the looks of it, I don't think you could get here. The calls to this already guard against this, so this is akin to a Debug.Assert.